sympy/physics/mechanics/functions.py

```diff
@@ -422,6 +447,10 @@ def find_dynamicsymbols(expression, exclude=None):
             raise TypeError("exclude kwarg must be iterable")
     else:
         exclude_set = set()
+    if isinstance(expression, Vector) and reference_frame is None :
+            raise TypeError("You must provide a reference frame when passing in a vector expression.")
```

Comment:
rather I would say, raise a `ValueError` error here not `TypeError`, see: https://docs.python.org/2/library/exceptions.html#exceptions.ValueError

Here since it is expected that user passes non-None `reference_frame`. Also use a better error message like:
`You must provide reference_frame when passing a vector expression, got %s." % reference_frame`

Reply:
Also I would write this as:
```
if isinstance(expression, Vector):
    if reference_frame is None:
        raise ValueError("You must provide reference_frame when passing a "
                         "vector expression, got %s." % reference_frame)
    else:
        expression = expression.to_matrix(reference_frame)
```

Also make sure character length on line is <= 79.